Python/sysmodule.c

```diff
@@ -917,7 +917,8 @@ sys_exit_impl(PyObject *module, PyObject *status)
 {
     /* Raise SystemExit so callers may catch it or clean up. */
     if (PyTuple_Check(status)) {
-        /* Make sure that tuples are not flattened during normalization. */
+        /* Make sure that tuples are not flattened during normalization
+         * due to the fast path for tuples in _PyErr_CreateException(). */
```

Comment:
It's not a "fast path" if removing it changes behaviour. We need to understand whether `_PyErr_CreateException` has a bug or not.

Reply:
I think it was designed as such for conveniency. It was added in 3a840972915ef05cb6d7c58df84100be05d9e35b but AFAICT, that was just refactoring of c0dc92af7d41ecf1d60f4ab29452d7179c039d8c (1997) which already had this fast path.